actionview/lib/action_view/helpers/cache_helper.rb

```diff
@@ -134,7 +134,7 @@ module CacheHelper
       #
       #   <%= render @notifications, cache: false %>
       def cache(name = {}, options = nil, &block)
-        if controller.perform_caching
+        if controller.try(:perform_caching)
```

Comment:
duh! You are completely right. I must be crazy. And after searching the entire codebase I saw some usage of `try`, so I think there is no need to avoid to use it.